examples/init-function/init_function.md

```diff
@@ -0,0 +1,12 @@
+# Init function
+
+The `init` function will be executed as the last function called in the program initialization phase. Uninitialized module level `final` or `non-final` variables can be initialized in this function.
+
+The `init` function does not accept arguments but it returns `error` or `nil`.
```

Comment:
We need to include this, `The init function must not be declared public; its return type must be a subtype of error? or `()`; it must have no parameters`

Reply:
Updated 4d4430791d8cfa1d2ab3f39ab24fbfe5202f325b